src/v2/models/AppState.js

```diff
@@ -35,21 +35,44 @@ export default Model.extend({
     factorProfile: {
       deps: ['factor'],
       fn (factor = {}) {
-        return factor.profile || {};
+        return factor && factor.profile || {};
       },
     },
     factorType: {
       deps: ['factor'],
       fn (factor = {}) {
-        return factor.factorType;
+        return factor && factor.factorType;
       },
     },
-    isTerminalState: {
+    isTerminalStep: {
       deps: ['terminal'],
       fn: function (terminal) {
-        return !_.isEmpty(terminal);
+        return !_.isEmpty(terminal) && terminal.value.length;
       },
     },
+    isAuthenticateStep: {
```

Comment:
we already dispatch logic in ViewFactory base on form-name. I think we could do Class extension instead of creating derived prop base on step. I'm imaging we'll have

- SelectFactorView
- SwitchFactorView

which could extend from some BaseFactorListView

Reply:
yeah, agree. This was a quick fix to get it working

Reply:
Thinking about this a bit more, we get back `select-factor` as the form name for both factor Required flow and factor enroll. Only thing different is that actual step `Authenticate` vs        
` Enroll` and we want to show different formTitle

Reply:
I thought we'll have `select-factor` and `challenge-factor` respectively?